### PR TITLE
Display Link email in Payment Methods page

### PR DESCRIPTION
**Pull request by annemirasol**
## Changes proposed in this Pull Request:
- In My Account > Payment Methods, display the Link email address for added context, the same way we display it in the Subscriptions and checkout pages.
- Tweak the checkout page display text, for consistency with Subscriptions and Payment Methods pages.

| Before | After |
|--------|-------|
| Screenshot 2024-12-06 at 11 27 42 AM | Screenshot 2024-12-06 at 11 27 56 AM |

## Testing instructions

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: includes/class-wc-stripe-payment-tokens.php [hunk 1]
```diff
@@ -392,7 +392,11 @@ public function get_account_saved_payment_methods_list_item( $item, $payment_tok
 				$item['method']['brand'] = esc_html__( 'Cash App Pay', 'woocommerce-gateway-stripe' );
 				break;
 			case WC_Stripe_Payment_Methods::LINK:
-				$item['method']['brand'] = esc_html__( 'Stripe Link', 'woocommerce-gateway-stripe' );
+				$item['method']['brand'] = sprintf(
```
Comment (mattallan) [inline includes/class-wc-stripe-payment-tokens.php:395]: This string should be escaped after the `sprintf()`.

Even though WooCommerce escapes this in their template `woocommerce/templates/myaccount/payment-methods.php` we can't guarantee this for all stores/themes that override this template 😞

File: changelog.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 *** Changelog ***
 
 = 9.0.0 - xxxx-xx-xx =
+* Tweak - Display email address for Link saved payment methods.
```
Comment (mattallan) [inline changelog.txt:4]: Sorry for missing this PR last week when on porter, but this changelog will needed to be moved to 9.1 🙏

**Review by mattallan — APPROVED** [review]
Nice fix @annemirasol and thanks for working on this!

I've left a few comments as part of code review but I've tested this and it's working well:

image

File: includes/class-wc-stripe-payment-tokens.php (hunk 1 above)
Comment (annemirasol) [reply includes/class-wc-stripe-payment-tokens.php:395]: Good call, escaping added in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3641/commits/ec8707dfbb7d0884af7ad95a4ebee3a9552a74da

File: changelog.txt (hunk 2 above)
Comment (annemirasol) [reply changelog.txt:4]: Moved changelog/readme entries to 9.1.0

**Pull request merged**